This patch is based on the upstream commit described below, 
modified to remove fuzz.

commit cf55ba1d20af34bf7bde10cb199f429aa06a9568
Author: Gabriel Majeri <gabriel.majeri6@gmail.com>
Date:   Fri Sep 27 15:52:25 2019 +0300

    Fix Clippy for main crate

Index: rust-packed-simd-0.3.3/src/api/minimal/ptr.rs
===================================================================
--- rust-packed-simd-0.3.3.orig/src/api/minimal/ptr.rs
+++ rust-packed-simd-0.3.3/src/api/minimal/ptr.rs
@@ -215,7 +215,7 @@ macro_rules! impl_minimal_p {
                     f,
                     "{}<{}>(",
                     stringify!($id),
-                    unsafe { crate::intrinsics::type_name::<T>() }
+                    crate::intrinsics::type_name::<T>()
                 )?;
                 for i in 0..$elem_count {
                     if i > 0 {
Index: rust-packed-simd-0.3.3/src/lib.rs
===================================================================
--- rust-packed-simd-0.3.3.orig/src/lib.rs
+++ rust-packed-simd-0.3.3/src/lib.rs
@@ -220,7 +220,8 @@
          clippy::must_use_candidate,
          // This lint is currently broken for generic code
          // See https://github.com/rust-lang/rust-clippy/issues/3410
-         clippy::use_self
+         clippy::use_self,
+         clippy::wrong_self_convention
 )]
 #![cfg_attr(test, feature(hashmap_internals))]
 #![deny(warnings, rust_2018_idioms, clippy::missing_inline_in_public_items)]
Index: rust-packed-simd-0.3.3/src/masks.rs
===================================================================
--- rust-packed-simd-0.3.3.orig/src/masks.rs
+++ rust-packed-simd-0.3.3/src/masks.rs
@@ -7,6 +7,7 @@ macro_rules! impl_mask_ty {
         pub struct $id($elem_ty);
 
         impl crate::sealed::Mask for $id {
+            #[inline]
             fn test(&self) -> bool {
                 $id::test(self)
             }
